pkg/tfvars/baremetal/baremetal.go

```diff
@@ -85,7 +85,11 @@ func TFVars(libvirtURI, bootstrapProvisioningIP, bootstrapOSImage, externalBridg
 
 		// Root device hints
 		rootDevice := make(map[string]interface{})
-		if profile.RootDeviceHints.HCTL != "" {
+
+		// host.RootHint overrides the root device hint in the profile
+		if host.RootHint != "" {
+			json.Unmarshal([]byte(host.RootHint), &rootDevice)
```

Comment:
no error handling here??

Reply:
this is seems really weird thing to support, to allow providing a JSON string of the end goal.